client/web: add support for zst precomppressed assets

raggi opened the pull request:
This will enable us to reduce the size of these embedded assets.

Updates tailscale/corp#20099

soniaappasamy reviewed (approved):
Tested locally to make sure things all still work okay

raggi commented:
I updated some more to do the build side, and tested against chrome with a local workspace setup, updating the build in the prebuilt repo locally. I caught a window size limit error with chrome, and that is also now fixed, so these are serving successfully now with precompressed zstd.

soniaappasamy reviewed (approved):
LGTM, once this gets in, we can run the following two actions (in this order) to update the prebuilt assets then make sure everything looks okay:

1. https://github.com/tailscale/web-client-prebuilt/actions/workflows/update-assets.yml
2. https://github.com/tailscale/tailscale/actions/workflows/update-webclient-prebuilt.yml

bradfitz commented on util/precompress/precompress.go:
```diff
@@ -63,13 +64,19 @@ type Options struct {
+			return f, nil
+		}
+	}
+	if tswebutil.AcceptsEncoding(r, "br") {
```
Maybe it's time to drop Brotli to make up for the increased cost/size of adding zstd?

Safari users can use gzip for now?

bradfitz commented on tsweb/tsweb.go:
```diff
@@ -94,25 +94,9 @@ func allowDebugAccessWithKey(r *http.Request) bool {
 
 // AcceptsEncoding reports whether r accepts the named encoding
 // ("gzip", "br", etc).
+// deprecated: use tswebutil.AcceptsEncoding instead.
```
//
// Deprecated: use ...
